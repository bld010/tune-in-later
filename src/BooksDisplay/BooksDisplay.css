/* body {
  background-color: #EEB7AC;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
} */

.genreLabel {
  z-index: 10;
  position: relative;
  top: 15px;
  font-size: 18px;
  left: 30px;
  opacity: 0.4;
  text-transform: lowercase;
  margin: 0 0 -20px 0;
  height: 30px;
}

section {
  background-color: #F0F0F0;
  width: 98%;
  margin: 20px;
  flex-direction: column;
  justify-content: flex-start;
  align-content: center;
  display: flex;
  overflow-x: scroll;
  overflow-y: hidden;
  border-radius: 8px;
}

section > div {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
}

article {
  margin: 15px; 
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
/*   border: 1px solid black; */
  background: white;
  width: 280px;
  height: 160px;
  border-radius: 8px;
  margin-top: 45px;
  box-shadow: 1px 1px 5px 5px lightgray;
}

div.img {
  height: 110px;
  width: 80px;
  margin: 20px;
  background: lightblue;
  
}

.bookInfo {
  width: 200px;
}

h2 {
  margin-bottom: 0px;
/*   padding: 0; */
}

h3 {
  margin-top: 0px;
}

h4 {
  margin-top: 35px;
}


h2, h3, h4 {
  font-family: sans-serif;
}

button {
  border: none;
  color: red;
  position: relative;
  left: 130px;
  top: -10px;
}

button:hover {
  cursor: pointer;
}