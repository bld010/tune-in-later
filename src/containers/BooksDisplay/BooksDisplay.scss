/* body {
  background-color: #EEB7AC;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
} */

.genreLabel {
  z-index: 10;
  position: relative;
  top: 15px;
  font-size: 18px;
  left: 30px;
  opacity: 0.4;
  text-transform: lowercase;
  margin: 0 0 -20px 0;
  height: 30px;
}

section {
  // background-color: #F0F0F0;
  background-color: rgba(255, 255, 255, .6);
  width: 98%;
  margin: 20px;
  flex-direction: column;
  justify-content: flex-start;
  align-content: center;
  display: flex;
  overflow-x: scroll;
  overflow-y: hidden;
  border-radius: 8px;
  will-change: transform;

  & div {
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
  }

  & h2 {
    margin-bottom: 0px;
    font-family: sans-serif;
  }
}
