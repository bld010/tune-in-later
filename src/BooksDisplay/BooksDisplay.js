import React from 'react';
import './BooksDisplay.css';

const BooksDisplay = ({books, sectionGenre}) => {
  const bookList = books.map(book => {
    const {artist, image, price, title, filterType} = book;
    return (
      <article key={title + price}>
        <div className="img">
          <img src={image} alt=''/>
        </div>
        <div className="bookInfo">
          <h2>{title}</h2>
          <h3>{artist}</h3>
          <h4>{filterType}</h4>
          <h5>{price}</h5>
          <button>Favorite</button>
        </div>
      </article>
    )
  });

  return(
    <section className='BooksDisplay'>
      <div>
        <h2 className="genreLabel">{sectionGenre}</h2>
      </div>
      <div>
        {bookList}
      </div>
    </section>
  )

}

export default BooksDisplay;