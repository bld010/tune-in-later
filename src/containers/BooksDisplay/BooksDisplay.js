import React from 'react';
import './BooksDisplay.css';
import { setCurrentUserFavorites } from '../../actions';
import { connect } from 'react-redux';

const BooksDisplay = ({books, sectionGenre}) => {
  console.log('books in booksdisplay', books)
  const bookList = books.map(book => {
    const {artist, image, price, title, filterType} = book;
    return (
      <article key={title + price}>
        <div className="img">
          <img src={image} alt=''/>
        </div>
        <div className="bookInfo">
          <h2>{title}</h2>
          <h3>{artist}</h3>
          <h4>{filterType}</h4>
          <h5>${price}</h5>
          <button>Favorite</button>
        </div>
      </article>
    )
  });

  return(
    <section className='BooksDisplay'>
      <div>
        <h2 className="genreLabel">{sectionGenre}</h2>
      </div>
      <div>
        {bookList}
      </div>
    </section>
  )
}

const mapStateToProps = state => ({
  currentUser : state.currentUser,
  favorites : state.favorites
})

const mapDispatchToProps = dispatch => ({
  setCurrentUserFavorites : favorites => dispatch(setCurrentUserFavorites(favorites))
});

export default connect(mapStateToProps, mapDispatchToProps)(BooksDisplay);