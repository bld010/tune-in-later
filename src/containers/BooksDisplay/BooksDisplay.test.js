import React from 'react';
import { shallow } from 'enzyme';
import { BooksDisplay, mapStateToProps } from './BooksDisplay';

describe('BooksDisplay', () => {

  global.Date.now = jest.spyOn(global.Date, 'now').mockImplementation(() => 12345);

  it('should match the snapshot with correct data passed in', () => {
   const mockBooks = [
     {
      artist:'Dante Alighieri',
      image:'http://someimage.com/',
      price:1,
      title:'A Tour of Turing',
      genre:'Horror',
      description:'Welcome to the secret 10th circle of Hell too chilling to be included in the Inferno!',
      date:'09/09/1320',
      filterType:'Horror',
      id:666
    },
     {
       artist:'David Gitlen',
       image:'http://davidgitlen.com/author.png',
       price:1,
       title:'Dante Was Right!',
       genre:'Memoir',
       description:'699 years to the day, somebody save me!',
       date:'09/09/2019',
       filterType:'History',
       id:111
    }
   ]
   const wrapper = shallow(<BooksDisplay 
    books={mockBooks}
    sectionGenre='Horror'
    favorites={[]}
   />);
   expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with correct data passed in if the books are from the user\'s favorites', () =>{
    const mockFavorites = [
      {
        author_name: 'Brianna DelValle',
        artwork_url: 'http://artwork.com/art.png',
        book_name: 'It\'s gonna be great!',
        primary_genre_name: 'Autobiography',
        description: 'It is gonna be great!',
        release_date: '09/09/2019',
        book_id: 222
      }
    ];

    const wrapper = shallow(<BooksDisplay 
      books={mockFavorites}
      sectionGenre='Favorites'
      favorites={mockFavorites}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return an object with appropriate data', () => {

      const mockState = {
        currentUser: {id: 1, name: 'David'},
        favorites: [{id: 12, title: 'Twelve'}, {id: 13, title: 'Thirteen'}],
        superfluous: 'Unnecessary data!'
      };
      const expected = {
        currentUser: { id: 1, name: 'David' },
        favorites: [{ id: 12, title: 'Twelve' }, { id: 13, title: 'Thirteen' }]
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);




    })
  })

})