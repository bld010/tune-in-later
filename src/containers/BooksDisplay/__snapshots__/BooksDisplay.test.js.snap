// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksDisplay should match the snapshot with correct data passed in 1`] = `
<section
  className="BooksDisplay"
>
  <div>
    <h2
      className="genreLabel"
    >
      Horror
    </h2>
  </div>
  <div>
    <Connect(Book)
      book={
        Object {
          "artist": "Dante Alighieri",
          "date": "09/09/1320",
          "description": "Welcome to the secret 10th circle of Hell too chilling to be included in the Inferno!",
          "filterType": "Horror",
          "genre": "Horror",
          "id": 666,
          "image": "http://someimage.com/",
          "price": 1,
          "title": "A Tour of Turing",
        }
      }
      key="8221770"
    />
    <Connect(Book)
      book={
        Object {
          "artist": "David Gitlen",
          "date": "09/09/2019",
          "description": "699 years to the day, somebody save me!",
          "filterType": "History",
          "genre": "Memoir",
          "id": 111,
          "image": "http://davidgitlen.com/author.png",
          "price": 1,
          "title": "Dante Was Right!",
        }
      }
      key="1370295"
    />
  </div>
</section>
`;

exports[`BooksDisplay should match the snapshot with correct data passed in if the books are from the user's favorites 1`] = `
<section
  className="BooksDisplay"
>
  <div>
    <h2
      className="genreLabel"
    >
      Favorites
    </h2>
  </div>
  <div>
    <Connect(Book)
      book={
        Object {
          "artist": "Brianna DelValle",
          "date": "09/09/2019",
          "description": "It is gonna be great!",
          "genre": "Autobiography",
          "id": 222,
          "image": "http://artwork.com/art.png",
          "title": "It's gonna be great!",
        }
      }
      key="2740590"
    />
  </div>
</section>
`;
